### PR TITLE
Hybrid oil heat pumps in ETM

### PR DESCRIPTION
**Pull request by KoenvanB**
This PR adds oil-based hybrid heat pumps to the front end. It includes the following additions:

1. Oil HHP slider for households and for buildings;
2. Threshold COP slider for oil HHP

The PR also includes the following changes to existing material:

- Updates of the `Demand response - behavior of hybrid heat pumps` slide text to refer to general HHPs (instead of only gas-based ones)
- Update of threshold COP slider texts for space heating & for hot water with additional explanation;
- Hides the `hybrid_heat_pump` table for electricity & gas shares (see [corresponding new issue](https://github.com/quintel/etmodel/issues/4237))
- label update for HHP gas & electricity consumer prices (closes [#4219](https://github.com/quintel/etmodel/issues/4219))

### CODE REVIEW

File: config/locales/interface/input_elements/en_demand_buildings.yml [hunk 1]
```diff
@@ -473,3 +473,37 @@ en:
         switches to hydrogen. The threshold value can be adjusted in the section 
         <a href=/scenario/flexibility/flexibility_net_load/demand-response-behavior-of-hybrid-heat-pumps
         >Demand response - behavior of hybrid heat pumps</a>.
+    buildings_space_heater_hybrid_crude_oil_heatpump_air_water_electricity_share:
+      title: Hybrid air heat pump (oil)
+      short_description:
+      description: |
+        This heater is a combination of an electric heat pump that draws its heat
```
Comment (mabijkerk) [inline config/locales/interface/input_elements/en_demand_buildings.yml:480]: I feel like the description for the HHP (or for all HHPs) is very long. I would prefer to keep the slider description short and to the point with relevant references to charts/other parts of the model, whereas the documentation may contain a more elaborate explanation of HHPs.

File: config/locales/interface/input_elements/en_demand_buildings.yml [hunk 2]
```diff
@@ -473,3 +473,37 @@ en:
...
+      short_description:
+      description: |
+        This heater is a combination of an electric heat pump that draws its heat 
+        from the outside air and an efficient conventional oil-fired boiler. <br/><br/>
+        This works as follows: <br/> 
+        The oil-fired part takes over all of the heating of the building on cold 
+        winter days, when heat demand peaks. Because of this clever trick, the heat 
+        pump has less impact on the electricity grid than an all-electric heat pump does. 
+        Smart use of hybrid crude oil heat pumps may therefore avoid the need to reinforce 
+        local power grids. <br/><br/>
```
Comment (mabijkerk) [inline config/locales/interface/input_elements/en_demand_buildings.yml:487]: Referring to the behaviour of the HHP as a "clever trick" does not seem very professional.

**Comment by mabijkerk** [comment]
For e.g., Germany or Northern Ireland, setting space heating 100% to hybrid (oil) heat pumps does not actually turn on the 'hybrid' part. By default, no ambient heat is consumed. The reason for this is that the threshold COP is rather high and decreasing it does lead to ambient heat consumption.

The functionaliteit is therefore verified, but in terms of validation I do wonder whether this is expected behaviour @KoenvanB? This effectively means that under current (or 2019) price conditions and temperatures, the hybrid heat pump will functions as an oil heater. Have you checked this with the client?

**Comment by mabijkerk** [comment]
> Hides the hybrid_heat_pump table for electricity & gas shares (see https://github.com/quintel/etmodel/issues/4237)

This table is still available through the slide text at Demand response - behavior of hybrid heat pumps":
Screenshot 2024-04-09 at 12 24 49

**Comment by KoenvanB** [comment]
I have 
- shortened the slide & slider texts by including part of it in the documentation (and inserted a reference to the documentation);
- removed the link to the hidden table;
- moved the oil HHP slider position to sit below the hydrogen HHP slider

**Comment by KoenvanB** [comment]
With the alterations mentioned adjustments this branch is ready to merge!

**Pull request merged**